### DISCUSSION TITLE
Airflow Scheduler is not working when DB connections is lost. It is not reinitiating the connection automatically after certain time.

### DISCUSSION THREAD

**Discussion by radiant0619** [apache/airflow#29130] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Airflow Scheduler is not working when DB connections is lost. It is not reinitiating the connection automatically after certain time.

### What you think should happen instead

It automatically try to connect the DB after some time

### How to reproduce

Just shut the DB and start the Airflow Scheduler service or vice versa

### Operating System

Centos 7

### Deployment

Other

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
This is correct and expected behaviour. Airlfow is not 100% fool-proof resistant to such events. You should make sure in your deployment that you detect such problems and restart the components if needed.

More explanation - Airflow is not the kind of software that is supposed to be 99.9X% relaiable due to it's batch/scheduling functionality. Systems that can automatically recover in all kinds of situations like that are considerably more complex and we simply chose not to handle such failures automatically for you. You should make sure your deployment implements all the probes and monitoring if you wish to recover from all similar sitiuations.

Converted it into discussion in case more discussion is needed.
